accumulo/src/main/scala/geotrellis/spark/io/accumulo/AccumuloLayerReader.scala

```diff
@@ -48,10 +48,14 @@ class AccumuloLayerReader(val attributeStore: AttributeStore)(implicit sc: Spark
 
     val queryKeyBounds = tileQuery(metadata)
 
-    val decompose = (bounds: KeyBounds[K]) =>
-      keyIndex.indexRanges(bounds).map { case (min, max) =>
-        new AccumuloRange(new Text(AccumuloKeyEncoder.long2Bytes(min)), new Text(AccumuloKeyEncoder.long2Bytes(max)))
-      }
+    val decompose = { (bounds: KeyBounds[K]) =>
+      if  (bounds == metadata.getComponent[Bounds[K]].get)
```

Comment:
I don't think we're able to write a layer with `EmptyKeyBounds`, so this seemed safe to me.